File: pydss/plot/DSSPlotImpl.py
from __pyjamas__ import (ARGERROR,)
from dss.plot.DSSPlot import (DSSPlot,)
# from java.awt.Color import (Color,)
# from java.util.ArrayList import (ArrayList,)
# from java.util.List import (List,)


class DSSPlotImpl(object, DSSPlot):
    DSSPlotObj = None
    addMarkerColor = None
    addMarkerCode = None
    addMarkerSize = None
    singlePhLineStyle = None
    threePhLineStyle = None

    class PlotType(object):
        AUTO_ADD_LOG_PLOT = 'AUTO_ADD_LOG_PLOT'
        CIRCUIT_PLOT = 'CIRCUIT_PLOT'
        GENERAL_DATA_PLOT = 'GENERAL_DATA_PLOT'
        GENERAL_CIRCUIT_PLOT = 'GENERAL_CIRCUIT_PLOT'
        MONITOR_PLOT = 'MONITOR_PLOT'
        DAISY_PLOT = 'DAISY_PLOT'
        METER_ZONES = 'METER_ZONES'
        LOAD_SHAPE = 'LOAD_SHAPE'
        TSHAPE = 'TSHAPE'
        PRICE_SHAPE = 'PRICE_SHAPE'
        PROFILE = 'PROFILE'
        _values = [AUTO_ADD_LOG_PLOT, CIRCUIT_PLOT, GENERAL_DATA_PLOT, GENERAL_CIRCUIT_PLOT, MONITOR_PLOT, DAISY_PLOT, METER_ZONES, LOAD_SHAPE, TSHAPE, PRICE_SHAPE, PROFILE]

        @classmethod
        def values(cls):
            return cls._enum_values[:]

    class PlotQuantity(object):
        VOLTAGE = 'VOLTAGE'
        CURRENT = 'CURRENT'
        POWER = 'POWER'
        LOSSES = 'LOSSES'
        CAPACITY = 'CAPACITY'
        NONE = 'NONE'
        _values = [VOLTAGE, CURRENT, POWER, LOSSES, CAPACITY, NONE]

        @classmethod
        def values(cls):
            return cls._enum_values[:]

    activeColorIdx = None
    colorArray = [None] * 17
    line = None
    transf = None
    bus1Idx = None
    bus2Idx = None
    generalCircuitPlotQuantity = None
    maxLineThickness = None
    plotType = None
    maxScale = None
    minScale = None
    dots = None
    labels = None
    showLoops = None
    showSubs = None
    # applies to meterZone plots only
    quantity = None
    objectName = None
    feederName = None
    valueIndex = None
    markerIdx = None
    # for general & autoAdd
    phasesToPlot = None
    # profile plot
    channels = None
    # for monitor plot
    bases = None
    # for monitor plot
    color1 = None
    color2 = None
    color3 = None
    # Tri-color plots
    triColorMax = None
    triColorMid = None
    maxScaleIsSpecified = None
    minScaleIsSpecified = None
    daisyBusList = None

    def __init__(self):
        # TODO Auto-generated constructor stub
        pass

    def doGeneralPlot(self):
        pass

    def doAutoAddPlot(self):
        pass

    def doTheDaisies(self):
        pass

    def doCircuitPlot(self):
        pass

    def doGeneralCircuitPlot(self):
        pass

    def doMeterZonePlot(self):
        pass

    def doMonitorPlot(self):
        pass

    def doProfilePlot(self):
        pass

    def markSubTransformers(self):
        """Misc support procedures"""
        pass

    def markTheTransformers(self):
        pass

    def doBusLabels(self, idx1, idx2):
        pass

    def doBusLabel(self, idx, busLabel):
        pass

    def labelBuses(self):
        pass

    def loadGeneralLineData(self):
        pass

    def setColorArray(self):
        pass

    def setMaxScale(self, *args):
        _0 = args
        _1 = len(args)
        if _1 == 0:
            pass # astStmt: [Stmt([]), None]
        elif _1 == 1:
            max, = _0
            self.maxScale = max
        else:
            raise ARGERROR(0, 1)

    def getColor(self):
        return 0

    def thickness(self):
        return 0

    def maxCurrent(self):
        return 0.0

    def nextColor(self):
        return None

    def quantityString(self):
        # private PenStyle style(int code) {
        # return null;
        # }
        return None

    def getAutoColor(self, scale):
        return None

    def getMarker(self, idx):
        return None

    def coordinateSame(self, i1, i2):
        return False

    def interpolateGradientColor(self, color1, color2, ratio):
        return None

    def setMaxLineThickness(self, value):
        pass

    def getMaxLineThickness(self):
        return self.maxLineThickness

    def getPlotType(self):
        return self.plotType

    def setPlotType(self, type):
        self.plotType = type

    def getMaxScale(self):
        return self.maxScale

    def getMinScale(self):
        return self.minScale

    def setMinScale(self, min):
        self.minScale = min

    def isDots(self):
        return self.dots

    def setDots(self, value):
        self.dots = value

    def isLabels(self):
        return self.labels

    def setLabels(self, lbl):
        self.labels = lbl

    def isShowLoops(self):
        return self.showLoops

    def setShowLoops(self, show):
        self.showLoops = show

    def isShowSubs(self):
        return self.showSubs

    def setShowSubs(self, show):
        self.showSubs = show

    def getQuantity(self):
        return self.quantity

    def setQuantity(self, qty):
        self.quantity = qty

    def getObjectName(self):
        return self.objectName

    def setObjectName(self, name):
        self.objectName = name

    def getFeederName(self):
        return self.feederName

    def setFeederName(self, name):
        self.feederName = name

    def getValueIndex(self):
        return self.valueIndex

    def setValueIndex(self, index):
        self.valueIndex = index

    def getMarkerIdx(self):
        return self.markerIdx

    def setMarkerIdx(self, idx):
        self.markerIdx = idx

    def getPhasesToPlot(self):
        return self.phasesToPlot

    def setPhasesToPlot(self, phases):
        self.phasesToPlot = phases

    def getChannels(self):
        return self.channels

    def setChannels(self, values):
        self.channels = values

    def getBases(self):
        return self.bases

    def setBases(self, values):
        self.bases = values

    def getColor1(self):
        return self.color1

    def setColor1(self, value):
        self.color1 = value

    def getColor2(self):
        return self.color2

    def setColor2(self, value):
        self.color2 = value

    def getColor3(self):
        return self.color3

    def setColor3(self, value):
        self.color3 = value

    def getTriColorMax(self):
        return self.triColorMax

    def setTriColorMax(self, max):
        self.triColorMax = max

    def getTriColorMid(self):
        return self.triColorMid

    def setTriColorMid(self, mid):
        self.triColorMid = mid

    def isMaxScaleIsSpecified(self):
        return self.maxScaleIsSpecified

    def setMaxScaleIsSpecified(self, isSpecified):
        self.maxScaleIsSpecified = isSpecified

    def isMinScaleIsSpecified(self):
        return self.minScaleIsSpecified

    def setMinScaleIsSpecified(self, isSpecified):
        self.minScaleIsSpecified = isSpecified

    def getDaisyBusList(self):
        return self.daisyBusList

    def setDaisyBusList(self, list):
        self.daisyBusList = list

    def execute(self):
        pass

    def setDefaults(self):
        pass

    def doLoadShapePlot(self, loadShapeName):
        pass

    def doTempShapePlot(self, tempShapeName):
        pass

    def doPriceShapePlot(self, priceShapeName):
        pass

    def doDI_Plot(self, caseName, caseYear, iRegisters, peakDay, meterName):
        pass

    def doCompareCases(self, caseName1, caseName2, whichFile, reg):
        pass

    def doYearlyCurvePlot(self, caseNames, whichFile, iRegisters):
        pass

    def doVisualizationPlot(self, element, quantity):
        pass

    @classmethod
    def setDSSPlotObj(cls, plotObj):
        cls.DSSPlotObj = plotObj

    @classmethod
    def getDSSPlotObj(cls):
        return cls.DSSPlotObj

    @classmethod
    def getAddMarkerColor(cls):
        return cls.addMarkerColor

    @classmethod
    def setAddMarkerColor(cls, color):
        cls.addMarkerColor = color

    @classmethod
    def getAddMarkerCode(cls):
        return cls.addMarkerCode

    @classmethod
    def setAddMarkerCode(cls, code):
        cls.addMarkerCode = code

    @classmethod
    def getAddMarkerSize(cls):
        return cls.addMarkerSize

    @classmethod
    def setAddMarkerSize(cls, size):
        cls.addMarkerSize = size

    @classmethod
    def getSinglePhLineStyle(cls):
        return cls.singlePhLineStyle

    @classmethod
    def setSinglePhLineStyle(cls, style):
        cls.singlePhLineStyle = style

    @classmethod
    def getThreePhLineStyle(cls):
        return cls.threePhLineStyle

    @classmethod
    def setThreePhLineStyle(cls, style):
        cls.threePhLineStyle = style

File: core/dss-core/src/main/java/com/ncond/dss/plot/impl/DSSPlotImpl.java
package com.ncond.dss.plot.impl;

import java.awt.Color;
import java.util.ArrayList;
import java.util.List;

import com.ncond.dss.common.CktElement;
import com.ncond.dss.delivery.LineObj;
import com.ncond.dss.delivery.TransformerObj;
import com.ncond.dss.plot.DSSPlot;

public class DSSPlotImpl implements DSSPlot {

	private static DSSPlot DSSPlotObj;

	private static int addMarkerColor;
	private static int addMarkerCode, addMarkerSize;
	private static int singlePhLineStyle;
	private static int threePhLineStyle;

	public enum PlotType {
		AUTO_ADD_LOG_PLOT,
		CIRCUIT_PLOT,
		GENERAL_DATA_PLOT,
		GENERAL_CIRCUIT_PLOT,
		MONITOR_PLOT,
		DAISY_PLOT,
		METER_ZONES,
		LOAD_SHAPE,
		TSHAPE,
		PRICE_SHAPE,
		PROFILE
	}

	public enum PlotQuantity {
		VOLTAGE, CURRENT, POWER, LOSSES, CAPACITY, NONE
	}

	private int activeColorIdx;
	private int[] colorArray = new int[17];
	private LineObj line;
	private TransformerObj transf;
	private int bus1Idx;
	private int bus2Idx;
	private String generalCircuitPlotQuantity;
	private int maxLineThickness;

	protected PlotType plotType;
	protected double maxScale;
	protected double minScale;
	protected boolean dots,
		labels,
		showLoops,  // applies to meterZone plots only
		showSubs;
	protected PlotQuantity quantity;
	protected String objectName,
		feederName;
	protected int valueIndex,
		markerIdx;  // for general & autoAdd

	protected int phasesToPlot;  // profile plot

	protected int[] channels;  // for monitor plot
	protected double[] bases;  // for monitor plot

	protected Color color1, color2, color3;

	/* Tri-color plots */
	protected double triColorMax, triColorMid;

	protected boolean maxScaleIsSpecified;
	protected boolean minScaleIsSpecified;

	protected List<String> daisyBusList;

	public DSSPlotImpl() {
		// TODO Auto-generated constructor stub
	}

	private void doGeneralPlot() {

	}

	private void doAutoAddPlot() {

	}

	private void doTheDaisies() {

	}

	private void doCircuitPlot() {

	}

	private void doGeneralCircuitPlot() {

	}

	private void doMeterZonePlot() {

	}

	private void doMonitorPlot() {

	}

	private void doProfilePlot() {

	}

	/** Misc support procedures */
	private void markSubTransformers() {

	}

	private void markTheTransformers() {

	}

	private void doBusLabels(int idx1, int idx2) {

	}

	private void doBusLabel(int idx, String busLabel) {

	}

	private void labelBuses() {

	}

	private void loadGeneralLineData() {

	}

	private void setColorArray() {

	}

	private void setMaxScale() {

	}

	private int getColor() {
		return 0;
	}

	private int thickness() {
		return 0;
	}

	private double maxCurrent() {
		return 0.0;
	}

	private Color nextColor() {
		return null;
	}

	private String quantityString() {
		return null;
	}

//	private PenStyle style(int code) {
//		return null;
//	}

	private Color getAutoColor(double scale) {
		return null;
	}

	private Byte getMarker(int idx) {
		return null;
	}

	private boolean coordinateSame(int i1, int i2) {
		return false;
	}

	private Color interpolateGradientColor(Color color1, Color color2, double ratio) {
		return null;
	}

	@Override
	public void setMaxLineThickness(int value) {

	}

	@Override
	public int getMaxLineThickness() {
		return maxLineThickness;
	}

	@Override
	public PlotType getPlotType() {
		return plotType;
	}

	@Override
	public void setPlotType(PlotType type) {
		plotType = type;
	}

	@Override
	public double getMaxScale() {
		return maxScale;
	}

	@Override
	public void setMaxScale(double max) {
		maxScale = max;
	}

	@Override
	public double getMinScale() {
		return minScale;
	}

	@Override
	public void setMinScale(double min) {
		minScale = min;
	}

	@Override
	public boolean isDots() {
		return dots;
	}

	@Override
	public void setDots(boolean value) {
		dots = value;
	}

	@Override
	public boolean isLabels() {
		return labels;
	}

	@Override
	public void setLabels(boolean lbl) {
		labels = lbl;
	}

	@Override
	public boolean isShowLoops() {
		return showLoops;
	}

	@Override
	public void setShowLoops(boolean show) {
		showLoops = show;
	}

	@Override
	public boolean isShowSubs() {
		return showSubs;
	}

	@Override
	public void setShowSubs(boolean show) {
		showSubs = show;
	}

	@Override
	public PlotQuantity getQuantity() {
		return quantity;
	}

	@Override
	public void setQuantity(PlotQuantity qty) {
		quantity = qty;
	}

	@Override
	public String getObjectName() {
		return objectName;
	}

	@Override
	public void setObjectName(String name) {
		objectName = name;
	}

	@Override
	public String getFeederName() {
		return feederName;
	}

	@Override
	public void setFeederName(String name) {
		feederName = name;
	}

	@Override
	public int getValueIndex() {
		return valueIndex;
	}

	@Override
	public void setValueIndex(int index) {
		valueIndex = index;
	}

	@Override
	public int getMarkerIdx() {
		return markerIdx;
	}

	@Override
	public void setMarkerIdx(int idx) {
		markerIdx = idx;
	}

	@Override
	public int getPhasesToPlot() {
		return phasesToPlot;
	}

	@Override
	public void setPhasesToPlot(int phases) {
		phasesToPlot = phases;
	}

	@Override
	public int[] getChannels() {
		return channels;
	}

	@Override
	public void setChannels(int[] values) {
		channels = values;
	}

	@Override
	public double[] getBases() {
		return bases;
	}

	@Override
	public void setBases(double[] values) {
		bases = values;
	}

	@Override
	public Color getColor1() {
		return color1;
	}

	@Override
	public void setColor1(Color value) {
		color1 = value;
	}

	@Override
	public Color getColor2() {
		return color2;
	}

	@Override
	public void setColor2(Color value) {
		color2 = value;
	}

	@Override
	public Color getColor3() {
		return color3;
	}

	@Override
	public void setColor3(Color value) {
		color3 = value;
	}

	@Override
	public double getTriColorMax() {
		return triColorMax;
	}

	@Override
	public void setTriColorMax(double max) {
		triColorMax = max;
	}

	@Override
	public double getTriColorMid() {
		return triColorMid;
	}

	@Override
	public void setTriColorMid(double mid) {
		triColorMid = mid;
	}

	@Override
	public boolean isMaxScaleIsSpecified() {
		return maxScaleIsSpecified;
	}

	@Override
	public void setMaxScaleIsSpecified(boolean isSpecified) {
		maxScaleIsSpecified = isSpecified;
	}

	@Override
	public boolean isMinScaleIsSpecified() {
		return minScaleIsSpecified;
	}

	@Override
	public void setMinScaleIsSpecified(boolean isSpecified) {
		minScaleIsSpecified = isSpecified;
	}

	@Override
	public List<String> getDaisyBusList() {
		return daisyBusList;
	}

	@Override
	public void setDaisyBusList(List<String> list) {
		daisyBusList = list;
	}

	@Override
	public void execute() {

	}

	@Override
	public void setDefaults() {

	}

	@Override
	public void doLoadShapePlot(String loadShapeName) {

	}

	@Override
	public void doTempShapePlot(String tempShapeName) {

	}

	@Override
	public void doPriceShapePlot(String priceShapeName) {

	}

	@Override
	public void doDI_Plot(String caseName, int caseYear, int[] iRegisters, boolean peakDay, String meterName) {

	}

	@Override
	public void doCompareCases(String caseName1, String caseName2, String whichFile, int reg) {

	}

	@Override
	public void doYearlyCurvePlot(ArrayList<String> caseNames, String whichFile, int[] iRegisters) {

	}

	@Override
	public void doVisualizationPlot(CktElement element, int quantity) {

	}

	public static void setDSSPlotObj(DSSPlot plotObj) {
		DSSPlotObj = plotObj;
	}

	public static DSSPlot getDSSPlotObj() {
		return DSSPlotObj;
	}

	public static int getAddMarkerColor() {
		return addMarkerColor;
	}

	public static void setAddMarkerColor(int color) {
		addMarkerColor = color;
	}

	public static int getAddMarkerCode() {
		return addMarkerCode;
	}

	public static void setAddMarkerCode(int code) {
		addMarkerCode = code;
	}

	public static int getAddMarkerSize() {
		return addMarkerSize;
	}

	public static void setAddMarkerSize(int size) {
		addMarkerSize = size;
	}

	public static int getSinglePhLineStyle() {
		return singlePhLineStyle;
	}

	public static void setSinglePhLineStyle(int style) {
		singlePhLineStyle = style;
	}

	public static int getThreePhLineStyle() {
		return threePhLineStyle;
	}

	public static void setThreePhLineStyle(int style) {
		threePhLineStyle = style;
	}

}
